File: src/main/java/org/veupathdb/lib/gradle/container/tasks/base/exec/BinExecAction.java
package org.veupathdb.lib.gradle.container.tasks.base.exec;

import org.gradle.api.tasks.Internal;
import org.jetbrains.annotations.NotNull;

import java.io.File;

/**
 * Bin Exec Action
 * <p>
 * Base class for build/bin dependency execution tasks.
 *
 * @since 1.1.0
 */
public abstract class BinExecAction extends ExecAction {
  @Override
  @Internal
  @NotNull
  protected File getWorkDirectory() {
    return log().getter(new File(RootDir, getOptions().getGlobalBinConfig().getBinDirectory()));
  }
}

File: src/main/java/org/veupathdb/lib/gradle/container/tasks/base/exec/BinExecAction.kt
package org.veupathdb.lib.gradle.container.tasks.base.exec

import org.gradle.api.tasks.Internal

import java.io.File

/**
 * Bin Exec Action
 * <p>
 * Base class for build/bin dependency execution tasks.
 *
 * @since 1.1.0
 */
abstract class BinExecAction : ExecAction() {
  @Internal
  override fun getWorkDirectory(): File {
    return log.getter(File(RootDir, options.utils.binDirectory))
  }
}
